deploy/helm/kubecf/templates/ingress.yaml

```diff
@@ -53,12 +53,27 @@ metadata:
     {{ $_ := set .Values.features.ingress.annotations "nginx.org/websocket-services" (printf "%s-router" .Values.deployment_name) }}
 {{ toYaml .Values.features.ingress.annotations | indent 4 }}
   labels:
+    {{ if not (hasKey .Values.features.ingress.labels "app.kubernetes.io/component") }}
     app.kubernetes.io/component: "ingress"
+    {{- end -}}
+    {{ if not (hasKey .Values.features.ingress.labels "app.kubernetes.io/instance") }}
     app.kubernetes.io/instance: {{ .Release.Name | quote }}
+    {{- end -}}
+    {{ if not (hasKey .Values.features.ingress.labels "app.kubernetes.io/managed-by") }}
     app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+    {{- end -}}
+    {{ if not (hasKey .Values.features.ingress.labels "app.kubernetes.io/name") }}
     app.kubernetes.io/name: {{ default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" | quote }}
+    {{- end -}}
+    {{ if not (hasKey .Values.features.ingress.labels "app.kubernetes.io/version") }}
     app.kubernetes.io/version: {{ default .Chart.Version .Chart.AppVersion | quote }}
+    {{- end -}}
+    {{ if not (hasKey .Values.features.ingress.labels "helm.sh/chart") }}
     helm.sh/chart: {{ printf "%s-%s" .Chart.Name (.Chart.Version | replace "+" "_") | quote }}
+    {{- end -}}
+{{ if .Values.features.ingress.labels }}
```

Comment:
Since you set a default empty `labels` key, you don't need to do this check here.

Reply:
We have to because `toYaml ` of an empty map will produce `{}` which is invalid syntax because of the other labels.

```yaml
  labels:
     app.kubernetes.io/component: "ingress" 
     helm.sh/chart: "chart-name-version"
     {}
```